src/test/regress/expected/multi_alter_table_add_constraints.out

```diff
@@ -0,0 +1,501 @@
+HINT:  You can issue each subcommand separately
+-- Tests for constraints without name
+-- Commands below should error out since constraints do not have the name 
+ALTER TABLE products ADD UNIQUE(product_no);
+ERROR:  cannot create constraint
+DETAIL:  Creating constraint without a name on a distributed table is currently not supported.
+ALTER TABLE products ADD PRIMARY KEY(product_no);
+ERROR:  cannot create constraint
+DETAIL:  Creating constraint without a name on a distributed table is currently not supported.
+ALTER TABLE products ADD CHECK(product_no <> 0);
+ERROR:  cannot create constraint
+DETAIL:  Creating constraint without a name on a distributed table is currently not supported.
+ALTER TABLE products ADD EXCLUDE USING btree (product_no with =);
+ERROR:  cannot create constraint
+DETAIL:  Creating constraint without a name on a distributed table is currently not supported.
+DROP TABLE products;
+-- Tests with transactions 
+BEGIN;
+CREATE TABLE products (
+    product_no integer,
+    name text,
+    price numeric
+);
+SELECT create_distributed_table('products', 'product_no');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+INSERT INTO products VALUES(1,'product_1', 5);
+-- Should error out since conflicts with the above single-shard data modification command.
+ALTER TABLE products ADD CONSTRAINT unn_pno UNIQUE(product_no);
+ERROR:  distributed DDL commands must not appear within transaction blocks containing single-shard data modifications
+COMMIT;
+BEGIN;
+CREATE TABLE products (
+    product_no integer,
+    name text,
+    price numeric,
+    discounted_price numeric
+);
+SELECT create_distributed_table('products', 'product_no');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- Add constraints
+ALTER TABLE products ADD CONSTRAINT unn_pno UNIQUE(product_no);
+ALTER TABLE products ADD CONSTRAINT check_price CHECK(price > discounted_price);
+ALTER TABLE products ALTER COLUMN product_no SET NOT NULL;
+ALTER TABLE products ADD CONSTRAINT p_key_product PRIMARY KEY(product_no);
+-- Single shard DML command can't be located in the same transaction with above commands.
+INSERT INTO products VALUES(1,'product_1', 10, 8);
+ERROR:  single-shard DML commands must not appear in transaction blocks which contain multi-shard data modifications
+COMMIT;
+-- Tests to check the effect of rollback
+CREATE TABLE products (
+    product_no integer,
+    name text,
+    price numeric,
+    discounted_price numeric
+);
+SELECT create_distributed_table('products', 'product_no');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+BEGIN;
+-- Add constraints (which will be rollbacked)
+ALTER TABLE products ADD CONSTRAINT unn_pno UNIQUE(product_no);
+ALTER TABLE products ADD CONSTRAINT check_price CHECK(price > discounted_price);
+ALTER TABLE products ADD CONSTRAINT p_key_product PRIMARY KEY(product_no);
+ROLLBACK;
+-- All of them should be equal to zero
```

Comment:
Instead of querying pg_constraint, you can just use

/d products and see there are no constraints.